pages/datarooms/index.tsx

```diff
@@ -0,0 +1,151 @@
+import DataroomCard from "@/components/datarooms/dataroom-card";
+import Skeleton from "@/components/Skeleton";
+import { PlusIcon } from "lucide-react";
+import { Separator } from "@/components/ui/separator";
+import AppLayout from "@/components/layouts/app";
+import { Button } from "@/components/ui/button";
+import { useState, useEffect, useRef } from "react";
+import LoadingSpinner from "@/components/ui/loading-spinner";
+import SelectDataroomTypeModal from "@/components/datarooms/select-dataroom-type-modal";
+import { DataroomWithFilesAndFolders } from "@/lib/types";
+import { useTeam } from "@/context/team-context";
+
+export interface DataroomWithFilesFoldersAndFilesCount
+  extends DataroomWithFilesAndFolders {
+  _count: {
+    files: number;
+  };
+}
+
+export default function Datarooms() {
+  const [datarooms, setDatarooms] = useState<
+    DataroomWithFilesFoldersAndFilesCount[] | undefined
+  >(undefined);
+  const [loading, setLoading] = useState<boolean>(false);
+  const teamInfo = useTeam();
+
+  //Fetch datarooms from backend
+  useEffect(() => {
+    (async () => {
+      if (teamInfo) {
+        const response = await fetch(
+          `/api/datarooms?teamId=${teamInfo?.currentTeam?.id}`,
+          {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          },
+        );
+
+        if (!response.ok) {
+          return;
+        }
+
+        const data = await response.json();
+        setDatarooms(data.datarooms);
+      }
+    })();
+  }, [teamInfo]);
```

Comment:
please use useSWR hook to fetch data! See other examples

Reply:
Done